src/cryptsetup/cryptsetup.c

```diff
@@ -352,6 +369,10 @@ static int parse_one_option(const char *option) {
                         arg_pkcs11_uri_auto = false;
                 }
 
+                /* If password-cache was not configured explicitly, default to no cache for PKCS#11 */
+                if (!arg_password_cache_set)
+                        arg_ask_password_flags &= ~(ASK_PASSWORD_ACCEPT_CACHED|ASK_PASSWORD_PUSH_CACHE);
```

Comment:
This works, but usually, handling inference with other options should be handled later.
So, please move this to `parse_crypt_config()`.